python/cudf/cudf/core/frame.py

```diff
@@ -550,6 +559,51 @@ def to_cupy(
         -------
         cupy.ndarray
         """
+        if (
+            self._num_columns > 1
+            and na_value is None
+            and self._columns[0].dtype.kind in {"i", "u", "f", "b"}
+            and all(
+                not col.nullable and col.dtype == self._columns[0].dtype
+                for col in self._columns
+            )
+        ):
+            if dtype is None:
+                dtype = np.dtype(self._columns[0].dtype)
```

Comment:
```suggestion
                dtype = self._columns[0].dtype
```
This should already be a `numpy.dtype`